halo/attest/keeper/query.go

```diff
@@ -49,6 +49,28 @@ func (k *Keeper) LatestAttestation(ctx context.Context, req *types.LatestAttesta
 	return &types.LatestAttestationResponse{Attestation: AttestationFromDB(att, sigs)}, nil
 }
 
+func (k *Keeper) EarliestAttestationInState(ctx context.Context, req *types.EarliestAttestationInStateRequest) (*types.EarliestAttestationInStateResponse, error) {
```

Comment:
Planning on adding unit tests in a separate PR, but feel free to say "no you have to do it now" :P